pkg/cgroup/cgroup.go

```diff
@@ -0,0 +1,292 @@
+package cgroup
+
+import (
+	"bufio"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/aquasecurity/tracee/pkg/mount"
+)
+
+type Cgroup interface {
+	initialize() error
+	Destroy() error
+	GetMountPoint() string
+	GetDefaultHierarchy() int
+	IsV1() bool
+}
+
+// cgroupv1
+
+func InitCgroupV1() (Cgroup, error) {
```

Comment:
So I think it would be best if we had here some `func New() (Cgroup, error)` and it would be responsible for decision making of which cgroup implementation to return, and that would be the preferred way to construct a cgroup object.

Reply:
That way in tracee.go itself there will be minimal exposure of internal decision making of which cgroups we need, just `t.cgroup, err := cgroup.New()` (obviously inside a `capabilities.Required` callback).